lib/galaxy/model/store/__init__.py

```diff
@@ -623,9 +626,17 @@ def handle_dataset_object_edit(dataset_instance, dataset_attrs):
                         else:
                             # Need a user to run library jobs to generate metadata...
                             pass
-                        self.app.datatypes_registry.set_external_metadata_tool.regenerate_imported_metadata_if_needed(
-                            dataset_instance, history, **regenerate_kwds
-                        )
+                        if self.app.datatypes_registry.set_external_metadata_tool:
+                            self.app.datatypes_registry.set_external_metadata_tool.regenerate_imported_metadata_if_needed(
+                                dataset_instance, history, **regenerate_kwds
+                            )
+                        else:
+                            # Try to set metadata directly. TODO: check efficiency of this?
```

Comment:
```suggestion
                            # Try to set metadata directly. @mvdbeek thinks we should only record the datasets
                            # that need (expensive) metadata generation and chain in another Celery task.
                            # This would let admins allocate worker / queue resources more efficiently. 
```